app/assets/javascripts/hqmf_util.js.coffee

```diff
@@ -1033,44 +1035,43 @@ selectConcurrent = (target, events) ->
   uniqueEvents((result for result in events when target.timeStamp().getTime() == result.timeStamp().getTime()))
 @selectConcurrent = selectConcurrent
 
-FIRST = (events) ->
-  result = []
-  result = selectConcurrent(events.sort(dateSortAscending)[0], events) if (events.length > 0)
-  applySpecificOccurrenceSubset('FIRST',hqmf.SpecificsManager.maintainSpecifics(result, events))
+# Common code for all subset operators
+applySubsetOperator = (operatorName, events, sortFunction, subsetIndex) ->
+  # If there is no specific context (that means we're being called from within specifics handling code), or
+  # there are no specifics involved, just perform the logical operator
+  unless events.specificContext && events.specificContext.hasSpecifics()
+    result = []
+    result = selectConcurrent(events.sort(sortFunction)[subsetIndex], events) if (events.length > subsetIndex)
+    hqmf.SpecificsManager.maintainSpecifics(result, events)
+    return result
+
+  # Specific occurrences are involved, which means that we have to return all the events that *might* satisfy
+  # the subset operator once specific occurrences are taken into account; start by calculating the specifics
+  events.specificContext = events.specificContext[operatorName]()
+
+  # Then, return the events after removing entries that *cannot possibly* satisfy the subset operator
+  hqmf.SpecificsManager.filterEventsAgainstSpecifics(events)
+
```

Comment:
Consider the following rephrasings for the comments.
Otherwise, looks good.

```
    # Start by calculating the specific context, which creats at least one row for event that satisfies the                                                                
    # subset operator for at least one of the specifics 
```
to
```
    # Start by calculating the specific context subset, which creates at least one row for each event that satisfies the                                                                
    # subset operator for at least one of the specifics 
```

and

```
   # Then, return the events after removing entries that *cannot possibly* satisfy the subset operator
```
to
```
   # Then, return only the events that can satisfy the subset operator for one or more specifics.
```